src/App.tsx

```diff
@@ -1,26 +1,226 @@
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import classNames from 'classnames';
+import { Footer } from './components/Footer';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>(todos);
+  const [title, setTitle] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [loadingTodoId, setLoadingTodoId] = useState<number | null>(null);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filter, setFilter] = useState('all');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const completedTodos = todos.filter(todo => todo.completed);
+  const notCompletedTodos = todos.filter(todo => !todo.completed);
+
+  // eslint-disable-next-line @typescript-eslint/no-shadow
+  function filterTodos(todos: Todo[], filter: string) {
+    const todosCopy = [...todos];
+
+    return todosCopy.filter(todo => {
+      switch (filter) {
+        case 'active':
+          return !todo.completed;
+
+        case 'completed':
+          return todo.completed;
+
+        default:
+          return true;
+      }
+    });
+  }
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [loading]);
+
+  useEffect(() => {
+    setFilteredTodos(() => filterTodos(todos, filter));
+  }, [filter, todos]);
+
+  useEffect(() => {
+    setErrorMessage('');
+    setLoading(true);
+
+    getTodos()
+      .then(setTodos)
+      .catch(error => {
+        setErrorMessage('Unable to load todos');
+
+        throw error;
+      })
+      .finally(() => setLoading(false));
+  }, []);
+
+  useEffect(() => {
+    const timerId = setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+
+    return () => clearTimeout(timerId);
+  }, [errorMessage]);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setErrorMessage('');
+    setTitle(event.target.value);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    setErrorMessage('');
+
+    if (title.trim().length === 0) {
+      setErrorMessage('Title should not be empty');
+
+      return;
+    }
+
+    setTempTodo({
+      id: 0,
+      userId: 2391,
+      title: title.trim(),
+      completed: false,
+    });
+
+    setLoading(true);
+    setLoadingTodoId(0);
+    createTodo({ title: title.trim(), completed: false })
+      .then(newTodo => {
+        setTempTodo(null);
+
+        setTodos(currentTodos => [...currentTodos, newTodo]);
+        setTitle('');
+      })
+      .catch(error => {
+        setErrorMessage('Unable to add a todo');
+        throw error;
+      })
+      .finally(() => {
+        setLoading(false);
+        setLoadingTodoId(null);
+        setTempTodo(null);
+      });
+  };
+
+  function onDeleteTodo(todoId: number) {
```

Comment:
It is important to stick to one way of creating functions, fix it everywhere.
```suggestion
  const onDeleteTodo = (todoId: number) => {
```